src/shogun/machine/GaussianProcessMachine.cpp

```diff
@@ -66,6 +66,8 @@ void GaussianProcessMachine::init()
 
 	SG_ADD(&m_method, "inference_method", "Inference method",
 	    ParameterProperties::HYPER);
+	add_callback_function(
```

Comment:
ẁhat is this for?

Reply:
https://github.com/shogun-toolbox/shogun/blob/5a8edb65af72f49c1c7e061746b86005cc9113de/src/shogun/classifier/GaussianProcessClassification.cpp#L53-L58
when we use `put("inducing_features", inducing_features)`, the constructor doesn't initialize the m_labels, so i add this add_callback_function to initialize `m_labels`.

Reply:
I see. Can't we do this in the `train` method?

Reply:
https://github.com/shogun-toolbox/shogun/blob/5a8edb65af72f49c1c7e061746b86005cc9113de/src/shogun/machine/Machine.cpp#L34-L42
it seems like that we can't do this in the `train_machine` method, as we must initialize m_labels before the `train_mechine` are called.

Reply:
Could we refactor the label passing in a way that the machine passes it to the inference method (in train) rather than the other way around?